### PR TITLE
Ports the black market

### PR DESCRIPTION
**Pull request by evermore9932**
## About The Pull Request
Ports the main pr and some follow up ones from tgstation
https://github.com/tgstation/tgstation/pull/48120
https://github.com/tgstation/tgstation/pull/52696
https://github.com/tgstation/tgstation/pull/52804
https://github.com/tgstation/tgstation/pull/56176
https://github.com/tgstation/tgstation/pull/57632
https://github.com/tgstation/tgstation/pull/57750
https://github.com/tgstation/tgstation/pull/57751
https://github.com/tgstation/tgstation/pull/57442
https://github.com/tgstation/tgstation/pull/49148 couldnt find the art for the hardsuit so it uses the ancient hardsuit as a placeholder
Moved some stuff from unsorted to turfs.dm in the process of this. Can move it back if thats unwanted.
First pr, hope I didnt mess this up

Issues: I tested it and there's a small problem with the ui. When you choose to buy something and the box for the shipping method appears, choosing either option will work but wont close that window, you have to click cancel afterwards. 

## Why It's Good For The Game
"Adds a money sink and it can be used to put content available if they don't have any other place.
And it can be later expanded to allow players to make their own shops to sell their goods."
You can now craft a device to access the black market with. Items bought with it can be shipped from space or teleported to a random spot on station, free for anyone to grab. You do get a headstart though.
It was in the projects list so it's gotta be good. Black market = good

## Changelog
:cl:
add: Adds the black market. You can craft a black market uplink to access it.
code: changed some code
refactor: Moved some procs from unsorted.dm to turfs.dm, similar to tg
/:cl:

### CODE REVIEW

**Comment by BrumalLake** [comment]
card link or proper pull links would be nice

**Comment by evermore9932** [comment]
It has links now

**Comment by Kapu1178** [comment]
so like. what *is* this.

File: beestation.dme [hunk 1]
```diff
@@ -29,11 +29,11 @@
 #include "code\__DEFINES\antagonists.dm"
 #include "code\__DEFINES\aquarium.dm"
 #include "code\__DEFINES\areas.dm"
-#include "code\__DEFINES\art.dm"
```
Comment (St0rmC4st3r) [inline beestation.dme]: Why is this removed?

Comment (evermore9932) [reply beestation.dme]: Is this related to the canvas stuff? I mightve had a slightly outdated fork. Does it need to be undone?

Comment (BrumalLake) [reply beestation.dme]: duh

Comment (St0rmC4st3r) [reply beestation.dme]: > Is this related to the canvas stuff? I mightve had a slightly outdated fork. Does it need to be undone?

Considering we got #5431 merged 21 hour ago? Yes, it does.
And undocumented changes are bad in general. Especially when they have nothing to do with the subject of the PR.

**Comment by evermore9932** [comment]
I've fixed another issue with canvas and enabled the blackmarket telepad. My bad. It should be working now.

**Comment by francinum** [comment]
I really hate that project.

File: code/modules/cargo/blackmarket/blackmarket_items/clothing.dm [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	item = /obj/item/clothing/head/chameleon/broken
+
+	price_min = 100
+	price_max = 200
+	stock_max = 2
+	availability_prob = 70
+
+/datum/blackmarket_item/clothing/combatmedic_suit
+	name = "Combat Medic hardsuit"
+	desc = "A discarded combat medic hardsuit, found in the ruins of a carpet bombed xeno hive. Definately used, but as sturdy as an anchor."
```
Comment (BrumalLake) [inline code/modules/cargo/blackmarket/blackmarket_items/clothing.dm]:
```suggestion
	desc = "A discarded combat medic hardsuit, found in the ruins of a carpet bombed xeno hive. Definitely used, but as sturdy as an anchor."
```

**Comment by francinum** [comment]
*No.*

Our economy is actually *SO LEGITIMATELY BROKEN* that the prices here are meaningless, and a nonzero amount of this gear is literal traitor gear, or stuff that should really not come out of thin air for the cost of almost nothing.

We'd essentially have to rebalance a shitload of the rest of the game around this.
🔑

**Comment by Crossedfall** [comment]
I think this was added back when Golden was a main server. I don't think any of this fits the experience we'd like to provide players, this feels very chaos & combat oriented. There's no reason why a normal crew member should have... Well... Really anything available via that uplink.

🔑
